blocks/block-edit/context.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * External dependencies
+ */
+import { createContext, createHigherOrderComponent } from '@wordpress/element';
+
+const { Consumer, Provider } = createContext( {
```

Comment:
I feel this context could also include what's current in the `BlockEdit` component's context right ? BlockList and canUserUseUnfilteredHTML

Reply:
Yes, we can consolidate it in this PR using new Context API unless it breaks more tests ...

Reply:
`BlocksList` looks complex because it depends on `createInnerBlockList` injected from `editor` ...
Another reason to merge those files to `editor` module :)